webrecorder/browsertrix
# Allow user to edit crawl template

(https://github.com/webrecorder/browsertrix-cloud/issues/144) Allows editing crawl template name and configuration & viewing previous versions. Also fixes crawl template list using old `/schedule` edit endpoint.

### Manual testing
1. Run app, go to a crawl template detail page
2. Click "Actions" -> "Change name" and save a different name. Verify name change is saved.
3. Click "Actions" -> "Edit schedule" and/or "Edit" in schedule section. Verify schedule change is saved.
4. Click "Actions" -> "Edit crawl configuration" and/or "Edit" in Configuration section. Verify there's a message indicating crawl template will be replaced at the top if config has crawls (see screenshot). Save and verify:
  - Browser URL updates to new config
  - "see previous version" link is shown under "Configuration" heading
  - Name and schedule has stayed the same
5. Click "see previous version". Verify you're taken to the old version, and links to new version exist (see screenshot)

### Screenshots

**New menu items in crawl template detail page:**
<img width="238" alt="Screen Shot 2022-02-21 at 2 51 14 PM" src="https://user-images.githubusercontent.com/4672952/155035848-e6c5c868-579c-49f5-89ef-e32836758895.png">

**Edit name modal:**
<img width="602" alt="Screen Shot 2022-02-21 at 2 25 24 PM" src="https://user-images.githubusercontent.com/4672952/155035913-9152b3d1-0a43-4fbe-a480-f4248b16244f.png">

**Edit config modal:**
<img width="617" alt="Screen Shot 2022-02-21 at 2 25 12 PM" src="https://user-images.githubusercontent.com/4672952/155035896-5b679ccd-d9f3-4089-b379-c7a6af1302bb.png">

**Link back to old config:**
<img width="585" alt="Screen Shot 2022-02-21 at 2 53 08 PM" src="https://user-images.githubusercontent.com/4672952/155035966-6e4b2dfd-fc72-4237-8d3c-bec373a650d7.png">

**Links to new config:**
<img width="1034" alt="Screen Shot 2022-02-21 at 2 54 05 PM" src="https://user-images.githubusercontent.com/4672952/155035982-ad22d824-2401-4060-bbfa-159195a93b55.png">
